harness/variables.go

```diff
@@ -0,0 +1,66 @@
+package harness
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	PluginErrorMessageKey = "PLUGIN_ERROR_MESSAGE"
+)
+
+// SetSecret sets a new secret by adding it to the output
+func SetSecret(name, value string) error {
+	return WriteEnvToFile(name, value)
+}
+
+// UpdateSecret updates an existing secret with a new value
+func UpdateSecret(name, value string) error {
+	return WriteEnvToFile(name, value)
+}
+
+// DeleteSecret removes a secret from the output
+func DeleteSecret(name string) error {
+	return WriteEnvToFile(name, "")
+}
+
+// SetError sets the error message and writes it to the DRONE_OUTPUT file
+func SetError(message string) error {
+	return WriteEnvToFile(PluginErrorMessageKey, message)
```

Comment:
```suggestion
func SetError(code string, message string) error {
	return WriteErrorToOutputFile(code, message)
```
introduce new func `WriteErrorToOutputFile` and use `PluginErrorMessageKey` inside it